packages/app/src/cli/services/app/config/push.ts

```diff
@@ -11,37 +11,45 @@ export interface Options {
   app: AppInterface
 }
 
+const DEFAULT_APP_VARIABLES = {
+  appProxy: null,
+  redirectUrls: [],
+  preferencesUrl: '',
+  gdprWebhooks: null,
+}
+
 export async function pushConfig(options: Options) {
   const token = await ensureAuthenticatedPartners()
   const mutation = PushConfig
 
   const {configuration} = options.app
   const configFileName = basename(options.app.configurationPath)
 
-  if (!configuration.client_id) {
-    abort(`${configFileName} does not contain a client_id.`)
-  }
-  const initialVariables = {
-    apiKey: configuration.client_id,
-    title: configuration.name,
-    applicationUrl: configuration.application_url,
-    redirectUrlAllowlist: configuration.redirect_url_allowlist,
-    requestedAccessScopes: configuration.requested_access_scopes,
-  }
+  if ('client_id' in configuration) {
```

Comment:
for values not contained in both app schema members, we'll need to always use [narrowing](https://www.typescriptlang.org/docs/handbook/2/narrowing.html)

"in" is a convenient way to do this